File: backend_server/models/driver_model.go
package models

type CreateDriverRequest struct {
	Car        string `bson:"car" json:"car" binding:"required"`
	Seats      int    `bson:"seats" json:"seats" binding:"required"`
	Experience int    `bson:"experience" json:"experience" binding:"required"`
}

type CreateDriverResponse struct {
	Car        string `bson:"car" json:"car" binding:"required"`
	Seats      int    `bson:"seats" json:"seats" binding:"required"`
	Experience int    `bson:"experience" json:"experience" binding:"required"`
	Email      string `bson:"email" json:"email" binding:"required"`
	Phone      string `bson:"phone" json:"phone" binding:"required"`
	Name       string `bson:"name" json:"name" binding:"required"`
}

type UpdateDriverRequest struct {
	Car        string `bson:"car" json:"car" binding:"required"`
	Seats      int    `bson:"seats" json:"seats" binding:"required"`
	Experience int    `bson:"experience" json:"experience" binding:"required"`
}

File: car_pooling_app/lib/model/driver/driver_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'driver_model.freezed.dart';
part 'driver_model.g.dart';

@freezed
class DriverModel with _$DriverModel {
  const factory DriverModel({
    required String name,
    required String car,  
    required String phone,
    required String email,
    required int seats,
    required int experience,
    required String gender,
  }) = _DriverModel;

  factory DriverModel.fromJson(Map<String, dynamic> json) =>
      _$DriverModelFromJson(json);
}